src/HDInsight/HDInsight/ManagementCommands/DisableAzureHDInsightAzureMonitorCommand.cs

```diff
@@ -0,0 +1,99 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.HDInsight.Commands;
+using Microsoft.Azure.Commands.HDInsight.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet("Disable", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "HDInsightAzureMonitor", DefaultParameterSetName = SetByNameParameterSet, SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class DisableAzureHDInsightAzureMonitorCommand : HDInsightCmdletBase
+    {
+        #region Input Parameter Definitions
+
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
```

Comment:
Please use appropriate parameter set names.

Reply:
> Please use appropriate parameter set names.

Hi @msJinLei , can we use this string? The other existing commands also use the parameter set: SetByNameParameterSet.
This means customer need to provide cluster name instead of providing resource id or object via pipeline.
We want to keep the same code style......

Reply:
You can name the parameter set as, for example, ByNameParameterSet = "ByNameParameterSet". SetByNameParameterSet leads to misunderstanding.

Reply:
```
src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Get-AzHDInsightHost.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Restart-AzHDInsightHost.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Restart-AzHDInsightHost.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightClusterDiskEncryptionKey.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightGatewayCredential.md:### SetByNameParameterSet (Default)
src/HDInsight/HDInsight/help/Set-AzHDInsightGatewayCredential.md:Parameter Sets: SetByNameParameterSet
src/HDInsight/HDInsight/help/Set-AzHDInsightGatewayCredential.md:Parameter Sets: SetByNameParameterSet
```
 "ByNamePrameterSet" seems to better in your case. You don't need to change all the cmdlets. Just change the cmdlets included in your PR.

Reply:
Changed per your comments. Thanks